src/cli.ts

```diff
@@ -304,9 +305,6 @@ async function run(options: Options) {
 		await tarExporter.saveToTar(todir, tmpdir, options.toTar, [options.toImage], options);
 	}
 	if (options.toRegistry) {
-		if (!options.token && allowInsecure == InsecureRegistrySupport.NO) {
```

Comment:
A similar check is already done on line 246.
If we decide to put it back here we should change `!options.token` to `!options.toToken` which created an error where images couldn't be pushed with only `options.toToken` set and `--allowInsecureRegistries` not set.